### ISSUE TITLE
Concurrency problem?

### ISSUE THREAD

**Issue by GilPalma** [jgauffin/Griffin.WebServer#14]
Hello! First of all, let me thank you for a quick and (mostly) painless solution for a problem that has plagued me for a while now.

I'm a college student but I'm also helping with a project as an intern at a software development company. This webserver has helped immensely but I've noticed something I cannot fix or understand. I cannot go into too much detail but the webserver setup in this project only works for the .NET webbrowser control and IE, simultaneously as well. Trying to access it from another browser (I've only tested this with Chrome, actually) leaves it waiting endlessly without any errors or any other message. Nothing even shows up in the Chrome developer console. After trying to repeat the process with a Python simple webserver (which is not portable so it's only useful for testing) everything works well.

Long story short: is there anything I can do to enable cross browser functionality, is it an incoming feature or is a fix not possible? Thank you for your time.

**Comment by jgauffin** [comment]
Just made a test case where I make two simultaneous `HttpWebRequest`. they both return an answer.

Can you create a small test case that reproduces your issue?

**Comment by jgauffin** [comment]
I've been able to reproduce it. Will take a look

**Comment by jgauffin** [comment]
Here is my test: https://github.com/jgauffin/Griffin.WebServer/blob/master/Source/Griffin.WebServer/Griffin.WebServer.Tests/GithubIssues/Issue14_should_work_with_concurrent_requests.cs

Feel free to adjust it to reproduce the issue

**Comment by GilPalma** [comment]
Without making any real changes, the project seems to have started working perfectly. I'm sorry for wasting your time but I might still find some use in your test later. Thank you so much for your speedy support!

**Comment by jgauffin** [comment]
no problem at all. another test is never wrong.
